cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/HttpRequestHandler.java

```diff
@@ -66,6 +66,7 @@
   private final Map<WrappedDiscoverable, MessageSender> discoveryLookup;
   private final List<ProxyRule> proxyRules;
 
+  private boolean exceptionHandled = false;
```

Comment:
It doesn't need to. Netty won't make concurrent call to a upstream handler instance. Also, no need to initialize to `false`. Java by default initialize fields.